### PR TITLE
feat(widget): option to hide safe banner

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Context: https://cowservices.slack.com/archives/C036G0J90BU/p1712267092933019

image

# To Test

1. Open widget configurator
2. Switch to standalone mode (or dapp-mode with connected Safe via WC)
3. Connect to Safe via WC
4. Setup sell ETH trade
- [ ] the banner should be displayed
5. Open browser console and execute `window.cowSwapWidgetParams = {banners: {hideSafeWebAppBanner: true}}`
6. Change some value in the configurator UI, for example: sell amount
7. Setup sell ETH trade again (if needed)
- [ ] the banner should not be displayed

### CODE REVIEW

**Comment by elena-zh** [comment]
hey @shoom3301 , I still see the banner for the Standalone mode when I execute the command: 
image
or 
image


Refreshing the page also does not help to see the changes. 

Could you please take a look into the issue once again?

**Comment by shoom3301** [comment]
Hi @elena-zh !

>Refreshing the page also does not help to see the changes.

You should not refresh the page, but you should do some change in the configurator UI, see 6th point in the test plan

File: libs/widget-lib/src/types.ts [hunk 1]
```diff
@@ -250,6 +250,19 @@ export interface CowSwapWidgetParams {
     orderError?: string | null
   }
 
+  banners?: {
+    /**
+     * Banner text: "Use Safe web app..."
+     *
+     * Conditions of the banner displaying:
```
Comment (alfetopito) [inline libs/widget-lib/src/types.ts]:
```suggestion
     * Conditions for displaying the banner:
```

File: libs/widget-lib/src/types.ts [hunk 2]
```diff
@@ -121,6 +121,47 @@ export interface CowSwapWidgetPalette {
...
+   * Alternatively, you can use a URL to a custom image file, or set to null to disable the image.
+   */
+  emptyOrders?: string | null
+}
+
+export interface CowSwapWidgetBanners {
+  /**
+   * Banner text: "Use Safe web app..."
+   *
+   * Conditions of the banner displaying:
```
Comment (alfetopito) [inline libs/widget-lib/src/types.ts:156]:
```suggestion
   * Conditions for displaying the banner:
```

File: libs/widget-lib/src/types.ts [hunk 3]
```diff
@@ -121,6 +121,47 @@ export interface CowSwapWidgetPalette {
...
+   *  - Safe-like app is connected to CoW Swap via WalletConnect
+   *  - Selling native token via Swap
```
Comment (alfetopito) [inline libs/widget-lib/src/types.ts:158]: Also:

- Sell token needs approval
- TWAP

**Review by elena-zh — APPROVED** [review]
Works, thanks!

File: apps/cowswap-frontend/src/modules/limitOrders/containers/LimitOrdersWarnings/index.tsx [hunk 4]
```diff
@@ -145,7 +147,7 @@ export function LimitOrdersWarnings(props: LimitOrdersWarningsProps) {
       {/*// TODO: must be replaced by <NotificationBanner>*/}
       {showHighFeeWarning && <SmallVolumeWarningBanner feeAmount={feeAmount} feePercentage={feePercentage} />}
       {showApprovalBundlingBanner && <BundleTxApprovalBanner />}
-      {showSafeWcBundlingBanner && <BundleTxSafeWcBanner />}
+      {showSafeWcBundlingBanner && !widgetParams?.banners?.hideSafeWebAppBanner && <BundleTxSafeWcBanner />}
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/limitOrders/containers/LimitOrdersWarnings/index.tsx:150]: Sure, fixed

File: libs/widget-lib/src/types.ts [hunk 5]
```diff
@@ -250,6 +250,19 @@ export interface CowSwapWidgetParams {
     orderError?: string | null
   }
 
+  banners?: {
+    /**
+     * Banner text: "Use Safe web app..."
+     *
+     * Conditions of the banner displaying:
+     *  - Safe-like app is connected to CoW Swap via WalletConnect
+     *  - Selling native token via Swap
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply libs/widget-lib/src/types.ts]: Hmmm, I don't see the banner usages in TWAP:

image

[the remark this answers is not in the document]
Comment (alfetopito) [reply libs/widget-lib/src/types.ts]: I see, those are the places where it's displayed!
On the other hand, I listed the conditions that require the Safe App.
For instance, you see something like that when you open the TWAP form without a Safe App, no?

**Pull request merged**